assets/js/react/components/common/AuthChecker/index.jsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { Button, CircularProgress, Fade, Typography } from '@material-ui/core';
+import { useIntl } from 'react-intl';
+import { useDispatch, useSelector } from 'react-redux';
+import styled from 'styled-components';
+
+import { displayLoginDialog } from '../../../actions/Auth';
+import isAuth from '../../../helpers/AuthHelper';
+
+import ErrorMessage from '../StatusMessage/ErrorMessage';
+
+// ====================
+
+const SpacedButton = styled(Button)`
+  ${({ theme }) => `
+    margin: ${theme.spacing(1)}px;
+`}
+`;
+
+const CenteredBlock = styled.div`
+  text-align: center;
+`;
+
+// ====================
+
+/**
+ * The AuthChecker checks if the user is authenticated:
+ * - if it's not, it displays an error message
+ * - if it is, it displays the provided component
+ *
+ * @param {PropTypes.element} errorMessageComponent
+ *    Component to display if the user is not authenticated
+ * @param {PropTypes.element} componentToDisplay
+ *    Component to display if the user is authenticated
+ */
+const AuthChecker = ({ errorMessageComponent, componentToDisplay }) => {
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const { formatMessage } = useIntl();
+
+  const [isUserAuth, setIsUserAuth] = useState(null);
+
+  const authState = useSelector((state) => state.auth);
```

Comment:
ca ressemble peut etre plus à une `feature` qu'un component

Reply:
> Si un user n'a pas les droits pour une feature, ce ne serait pas plus UX friendly qu'elle soit
> 
>     * Disable avec `Tooltip`

J'ai préféré proposer le log in histoire que l'utilisateur ne soit pas bloquée dans une impasse en terme d'actions réalisables. Avec un tooltip, il aura juste l'info qu'il doit se logguer, sans avoir la possibilité de le faire facilement.

>     * Cacher la feature
>
C'est ce que je fais déjà non ? Ou j'ai mal compris ^^ ?

Reply:
J'ai hésité, ça me paraît bien en feature aussi, je vais le bouger